src/packages/engine-core/src/NodeEngine.ts

```diff
@@ -166,6 +172,12 @@ export class NodeEngine {
     )
     this.engineEndpoint = engineEndpoint
 
+    if (useUds && process.platform !== 'win32') {
+      this.socketPath = `/tmp/prisma-${getRandomString()}.sock`
```

Comment:
What are the chances that the filesystem refuse access to `/tmp/` or if it's read only? I guess the socket would not initialize?

Should we check that first maybe like this?

```js
try {
	// R_OK | Flag indicating that the file can be read by the calling process.
	// W_OK | Flag indicating that the file can be written by the calling process.
    fs.accessSync(path.dirname(this.socketPath), fs.constants.R_OK | fs.constants.W_OK)
}
catch (err) {
    // err
}
```

Reply:
I'm not aware of any case, in which `/tmp` is not writable.
I would personally keep it simple for now.